criu/cr-restore.c

```diff
@@ -447,15 +447,10 @@ static int restore_native_sigaction(int sig, SaEntry *e)
 	ASSIGN_TYPED(act.rt_sa_handler, decode_pointer(e->sigaction));
 	ASSIGN_TYPED(act.rt_sa_flags, e->flags);
 	ASSIGN_TYPED(act.rt_sa_restorer, decode_pointer(e->restorer));
-#ifdef CONFIG_MIPS
+	memcpy(act.rt_sa_mask.sig, &e->mask, sizeof(e->mask));
+#ifdef ARCH_HAS_SIG_MASK_EXTENDED
 	e->has_mask_extended = 1;
```

Comment:
another issue here (it comes from the original commit https://github.com/checkpoint-restore/criu/commit/afe90627e276e9e68cdf4872cae8fa3c5c637914) is that we assigning `e->has_mask_extended` here. We should check it to be equal `1` instead, like this:
`BUG_ON(!e->has_mask_extended)`
which means "fail if ARCH_HAS_SIG_MASK_EXTENDED but sa.proto image contains no extended mask".